### ISSUE TITLE
Extract printers from `standalone` bundle

### ISSUE THREAD

**Issue by sosukesuzuki** [prettier/prettier#13256]
Now, `standalone` bundle includes printers for languages.

Ideally printers should be bundled as `./printers/*.js` like parsers.

What do you think?

**Comment by fisker** [comment]
To hard for user to use.


```js
import * as prettier from "prettier/standalone";
import prettierBabelParser from "prettier/parser-babel";
import prettierEstreePrinter from "prettier/printer-estree";

await prettier.format(`foo`, {plugins: [prettierBabelParser, prettierEstreePrinter]})
```

**Comment by fisker** [comment]
Mentioned here https://github.com/prettier/prettier/issues/12144#issuecomment-1023143020

**Comment by fisker** [comment]
But if we add "loadPlugin" logic (auto load via network?) to standalone version, it will work.

**Comment by thorn0** [comment]
How about bundling printers in one file with parsers for languages that have only one parser?

**Comment by fisker** [comment]
We can do that, maybe we'll only need one extra `estree` printer. 
But I guess the final bundle will be bigger, they can't reuse `doc` module.

**Comment by thorn0** [comment]
One solution is that if a function is passed as a plugin it's called with things like `doc` passed as arguments. So these bundles will export function and the core will know how to handle that.

**Comment by thorn0** [comment]
It'd also kind of address https://github.com/prettier/prettier/issues/10156
Plugins wouldn't need to depend on the Prettier module anymore.

**Comment by fisker** [comment]
But core printers already imported `doc` module, requires lots of changes.

**Comment by thorn0** [comment]
I believe this can be solved on the bundler level, without changing the code of printers.

**Comment by thorn0** [comment]
This reminded me of the AMD module format, bundlers know how to do that:
```js
define(["doc"], (doc) => {
  // ...
});
```
This output is very close to what we need.

**Comment by fisker** [comment]
@thorn0 @sosukesuzuki 

I tried to bundle the printer into graphql plugin, turns it's not too bad.

```text
**Size Change:** -8.08 kB (0%) 

**Total Size:** 9.02 MB

| Filename | Size | Change |  |
| :--- | :---: | :---: | :---: |
| `./dist/index.mjs` | 1.21 MB | -15.6 kB (-1%) |  |
| `./dist/plugins/graphql.js` | 46.1 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/graphql.mjs` | 45.3 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/standalone.js` | 337 kB | -7.84 kB (-2%) |  |
| `./dist/standalone.mjs` | 337 kB | -7.84 kB (-2%) |  |
```

The 7.84 -> 11.6 because we can't reuse `doc` module, let do this? We can find a way to reuse `doc` later.

For the js printers, we need add a new `plugins/estree.mjs` plugin, or leave it in core for now.

**Comment by fisker** [comment]
I've done moving everything except js/json.

<details>
<summary>Size Changes</summary>

// graphql

**Size Change:** -8.07 kB (0%) 

**Total Size:** 9.02 MB

| Filename | Size | Change |  |
| :--- | :---: | :---: | :---: |
| `./dist/index.mjs` | 1.21 MB | -15.6 kB (-1%) |  |
| `./dist/plugins/graphql.js` | 46.1 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/graphql.mjs` | 45.3 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/standalone.js` | 337 kB | -7.84 kB (-2%) |  |
| `./dist/standalone.mjs` | 337 kB | -7.84 kB (-2%) |  |

// yaml

**Size Change:** -26.9 kB (0%) 

**Total Size:** 9 MB

| Filename | Size | Change |  |
| :--- | :---: | :---: | :---: |
| `./dist/index.mjs` | 1.18 MB | -45.1 kB (-4%) |  |
| `./dist/plugins/graphql.js` | 46.1 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/graphql.mjs` | 45.3 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/yaml.js` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/plugins/yaml.mjs` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/standalone.js` | 325 kB | -20.4 kB (-6%) | ✅ |
| `./dist/standalone.mjs` | 324 kB | -20.4 kB (-6%) | ✅ |

// html

**Size Change:** -84.9 kB (-1%) 

**Total Size:** 8.94 MB

| Filename | Size | Change |  |
| :--- | :---: | :---: | :---: |
| `./dist/index.mjs` | 1.11 MB | -114 kB (-9%) | ✅ |
| `./dist/plugins/graphql.js` | 46.1 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/graphql.mjs` | 45.3 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/html.js` | 143 kB | +34 kB (+31%) | 🚨 |
| `./dist/plugins/html.mjs` | 142 kB | +34 kB (+31%) | 🚨 |
| `./dist/plugins/yaml.js` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/plugins/yaml.mjs` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/standalone.js` | 296 kB | -49.2 kB (-14%) | 👏 |
| `./dist/standalone.mjs` | 296 kB | -49.2 kB (-14%) | 👏 |

// glimmer

**Size Change:** -99.8 kB (-1%) 

**Total Size:** 8.92 MB

| Filename | Size | Change |  |
| :--- | :---: | :---: | :---: |
| `./dist/index.mjs` | 1.09 MB | -136 kB (-11%) | 👏 |
| `./dist/plugins/glimmer.js` | 133 kB | +13.2 kB (+11%) | ⚠️ |
| `./dist/plugins/glimmer.mjs` | 132 kB | +13.2 kB (+11%) | ⚠️ |
| `./dist/plugins/graphql.js` | 46.1 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/graphql.mjs` | 45.3 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/html.js` | 143 kB | +34 kB (+31%) | 🚨 |
| `./dist/plugins/html.mjs` | 142 kB | +34 kB (+31%) | 🚨 |
| `./dist/plugins/yaml.js` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/plugins/yaml.mjs` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/standalone.js` | 287 kB | -58.6 kB (-17%) | 👏 |
| `./dist/standalone.mjs` | 286 kB | -58.6 kB (-17%) | 👏 |

// markdown

**Size Change:** -89.1 kB (-1%) 

**Total Size:** 8.93 MB

| Filename | Size | Change |  |
| :--- | :---: | :---: | :---: |
| `./dist/index.mjs` | 1.04 MB | -184 kB (-15%) | 👏 |
| `./dist/plugins/glimmer.js` | 133 kB | +13.2 kB (+11%) | ⚠️ |
| `./dist/plugins/glimmer.mjs` | 132 kB | +13.2 kB (+11%) | ⚠️ |
| `./dist/plugins/graphql.js` | 46.1 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/graphql.mjs` | 45.3 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/html.js` | 143 kB | +34 kB (+31%) | 🚨 |
| `./dist/plugins/html.mjs` | 142 kB | +34 kB (+31%) | 🚨 |
| `./dist/plugins/markdown.js` | 157 kB | +51.8 kB (+49%) | 🚨 |
| `./dist/plugins/markdown.mjs` | 156 kB | +51.8 kB (+50%) | 🆘 |
| `./dist/plugins/yaml.js` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/plugins/yaml.mjs` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/standalone.js` | 264 kB | -81.2 kB (-24%) | 🎉 |
| `./dist/standalone.mjs` | 264 kB | -81.2 kB (-24%) | 🎉 |

// css

**Size Change:** -125 kB (-1%) 

**Total Size:** 8.9 MB

| Filename | Size | Change |  |
| :--- | :---: | :---: | :---: |
| `./dist/index.mjs` | 995 kB | -232 kB (-19%) | 👏 |
| `./dist/plugins/glimmer.js` | 133 kB | +13.2 kB (+11%) | ⚠️ |
| `./dist/plugins/glimmer.mjs` | 132 kB | +13.2 kB (+11%) | ⚠️ |
| `./dist/plugins/graphql.js` | 46.1 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/graphql.mjs` | 45.3 kB | +11.6 kB (+34%) | 🚨 |
| `./dist/plugins/html.js` | 143 kB | +34 kB (+31%) | 🚨 |
| `./dist/plugins/html.mjs` | 142 kB | +34 kB (+31%) | 🚨 |
| `./dist/plugins/markdown.js` | 157 kB | +51.8 kB (+49%) | 🚨 |
| `./dist/plugins/markdown.mjs` | 156 kB | +51.8 kB (+50%) | 🆘 |
| `./dist/plugins/postcss.js` | 154 kB | +29.2 kB (+23%) | 🚨 |
| `./dist/plugins/postcss.mjs` | 153 kB | +29.3 kB (+24%) | 🚨 |
| `./dist/plugins/yaml.js` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/plugins/yaml.mjs` | 124 kB | +17.9 kB (+17%) | ⚠️ |
| `./dist/standalone.js` | 241 kB | -104 kB (-30%) | 🎉 |
| `./dist/standalone.mjs` | 240 kB | -104 kB (-30%) | 🎉 |

</details>

**Comment by fisker** [comment]
Fixed by #14354 and #14570
